code-challenges.test.js

```diff
@@ -9,51 +12,207 @@
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.354 s, estimated 1 s
+// Ran all test suites.
+    
+
+ 
+
+    
+    
+
+
+
+
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+describe("nounFinder1", () => {
+      it("takes in an array of words and returns an array with all the words capitalized", () => { 
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
+    
+        expect(nounfinder1(randomNouns1)).toEqual( ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(nounfinder1(randomNouns2)).toEqual( ["Temperature", "Database", "Chopsticks", "Mango"])
+      })
+    })
+
+
+// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+// ● nounFinder › takes in an array of words and returns an array with all the words capitalized
+// ReferenceError: nounFinder is not defined
+
 // b) Create the function that makes the test pass.
+// Psuedocode: create a function named Nounfinder1 and Nounfinder2
+// parameter will be an array named array
+// Input : we want to input an array of strings
+//  Output: we expect to return an array of strings with all of the words capitalized within. 
+// create a function that takes in an array of words and returns array with all the words capitalized
+const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+// // declare a function that takes in an array
+const Nounfinder1 = (array) => {
+    //  return the array output from the map action
+    return array.map ( (value) => {
+        // with each value access the first index and capitalize the letter then concatenate the remainder of the word
+        return value[0].toUppercase() + value.conCat(1)
+    } )
+}
+// I keep getting the     ReferenceError: nounfinder1 is not defined, I've tried a variety of different syntax's to tackle this problem and I believe I'm over thinking the process. I understand the overall concept of testing and creating higher order functions, however I'm struggling with implementation.  I would like for someone to go over the logic behind this process a bit more, perhaps in this weeks office hours.  I've ran the test about 10 different times over the last 3 hours, so at this point I will opt to move on from this problem and tackle another . 
```

Comment:
It is always the littlest things ... Let's take a look at line 155 and then scroll up to line 128. Simple fix but an important one! I want you to get this working and then get it back to me. The error is telling you something important

Reply:
I spent a lot of time looking over this and I still can't get it to work,  I know the error is telling me that Nounfinder1 is undefined but every change I make still gets that error. I'm really struggling with this one.

Reply:
<img width="1114" alt="Screenshot 2023-03-01 at 6 18 26 PM" src="https://user-images.githubusercontent.com/121983774/222314076-2e627a37-f2c6-4af4-b5f0-45b1a261051a.png">

Reply:
<img width="1117" alt="Screenshot 2023-03-01 at 6 56 03 PM" src="https://user-images.githubusercontent.com/121983774/222319414-658597c4-17f4-46be-b61a-f0ac83f46e1e.png">